src/components/FeatureFlag.tsx

```diff
@@ -8,6 +9,31 @@ export type FeatureFlagComponentProps = {
     children: ReactNode,
 };
 
+const checkFlagStrategies = (flag: FlagValue, userId: string | undefined) => {
```

Comment:
would be nice to split up each strategy in a separate file to keep it maintainable when more and more complex strategies will be added